heron/ckptmgr/tests/java/com/twitter/heron/ckptmgr/CheckpointManagerServerTest.java

```diff
@@ -0,0 +1,387 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Message;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.twitter.heron.common.basics.NIOLooper;
+import com.twitter.heron.common.basics.SysUtils;
+import com.twitter.heron.common.network.HeronClient;
+import com.twitter.heron.common.network.HeronSocketOptions;
+import com.twitter.heron.common.network.StatusCode;
+import com.twitter.heron.proto.ckptmgr.CheckpointManager;
+import com.twitter.heron.proto.system.PhysicalPlans;
+import com.twitter.heron.spi.statefulstorage.Checkpoint;
+import com.twitter.heron.spi.statefulstorage.IStatefulStorage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class CheckpointManagerServerTest {
+
+  private static final String TOPOLOGY_NAME = "topology_name";
+  private static final String TOPOLOGY_ID = "topology_id";
+  private static final String CHECKPOINT_MANAGER_ID = "ckptmgr_id";
+  private static final String INSTANCE_ID = "instance_id";
+  private static final String STMGR_ID = "stmgr_id";
+  private static final int TASK_ID = 1;
+  private static final int COMPONENT_INDEX = 1;
+  private static final String CHECKPOINT_ID = "checkpoint_id";
+  private static final String COMPONENT_NAME = "component_name";
+  private static final byte[] BYTES = "checkpoint manager server test bytes".getBytes();
+
+  private static final String SERVER_HOST = "127.0.0.1";
+  private static int serverPort;
+
+  private static CheckpointManager.InstanceStateCheckpoint instanceStateCheckpoint;
+  private static CheckpointManager.SaveInstanceStateRequest saveInstanceStateRequest;
+  private static CheckpointManager.GetInstanceStateRequest getInstanceStateRequest;
+  private static CheckpointManager.CleanStatefulCheckpointRequest cleanStatefulCheckpointRequest;
+  private static CheckpointManager.RegisterStMgrRequest registerStmgrRequest;
+  private static CheckpointManager.RegisterTMasterRequest registerTMasterRequest;
+
+  private static PhysicalPlans.Instance instance;
+
+  private CheckpointManagerServer checkpointManagerServer;
+  private SimpleCheckpointManagerClient simpleCheckpointManagerClient;
+
+  private NIOLooper serverLooper;
+  private NIOLooper clientLooper;
+  private ExecutorService threadPools;
+  private CountDownLatch finishedSignal;
+
+  private IStatefulStorage backendStorage;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    PhysicalPlans.InstanceInfo info = PhysicalPlans.InstanceInfo.newBuilder()
```

Comment:
refactored the code so that `LocalFileSystemStorageTest` and `HDFSStorageTest` shares same setup code.

not making this change to `CkeckpointManagerServerTest` code. Logically, it's separate from the other two, so I prefer also keep them separate on the code part.